pkg/workloadmeta/collectors/internal/ecsfargate/ecsfargate.go

```diff
@@ -159,6 +159,10 @@ func (c *collector) parseTaskContainers(
 		seen[entityID] = struct{}{}
 
 		image, err := workloadmeta.NewContainerImage(container.Image)
+
+		// Add imageID to image
+		image.ID = container.ImageID
```

Comment:
In order to ensure we don’t forget to set the image ID in one collector, it would be cleaner to fix the `workloadmeta.NewContainerImage(…)` function in [`pkg/workloadmeta/types.go`](https://github.com/DataDog/datadog-agent/blob/77a596931aa9faa67ae398be252bfcd4519e7338/pkg/workloadmeta/types.go#L280-L285) to ensure it fills all the `workloadmeta.ContainerImage` fields, including the `ID` one.

```diff
--- a/pkg/workloadmeta/types.go
+++ b/pkg/workloadmeta/types.go
@@ -278,8 +278,9 @@ type ContainerImage struct {
 }
 
 // NewContainerImage builds a ContainerImage from an image name
-func NewContainerImage(imageName string) (ContainerImage, error) {
+func NewContainerImage(imageID, imageName string) (ContainerImage, error) {
        image := ContainerImage{
+               ID:      imageID,
                RawName: imageName,
                Name:    imageName,
        }
```

```suggestion
		image, err := workloadmeta.NewContainerImage(container.ImageID, container.Image)
```